pkg/api/login.go

```diff
@@ -82,10 +82,10 @@ func (hs *HTTPServer) LoginView(c *models.ReqContext) {
 			if err := validateRedirectTo(redirectTo); err != nil {
 				viewData.Settings["loginError"] = err.Error()
 				c.HTML(200, getViewIndex(), viewData)
-				c.SetCookie("redirect_to", "", -1, setting.AppSubUrl+"/")
+				hs.deleteCookie(c.Resp, "redirect_to", http.SameSiteDefaultMode)
```

Comment:
All of the `hs.deleteCookie` in this file are providing the wrong same sate attribute, should be taken from Grafana settings.